overlord/state/task.go

```diff
@@ -46,20 +48,26 @@ func newTask(state *State, id, kind, summary string) *Task {
 }
 
 type marshalledTask struct {
-	ID      string                      `json:"id"`
-	Kind    string                      `json:"kind"`
-	Summary string                      `json:"summary"`
-	Data    map[string]*json.RawMessage `json:"data"`
+	ID          string                      `json:"id"`
+	Kind        string                      `json:"kind"`
+	Summary     string                      `json:"summary"`
+	Status      Status                      `json:"status"`
+	CurProgress int                         `json:"cur_progess"`
```

Comment:
Yes, I'd go with one of these:

* `"progress": {"current": 0, "total", 1}`
* `current-progress` and `total-progress`
* `current` and `total`